### ISSUE TITLE
reportUnneccearyComparison false positive since 1.1.297 with `match` on `int | float | X`

### ISSUE THREAD

**Issue by parched** [microsoft/pyright#4800]
**Screenshots or Code**
```python

class V:
    pass

def format_(
    value: int | float | V,
) -> str:
    match value:
        case float():
            return "1"
        case int(): # triggers 'Pattern will never be matched for subject type "V"'
            return "2"
        case V():
            return "3"
```

**VS Code extension or command-line**
pyright command line 1.1.295 is fine, 1.1.296 gives this error.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.300, which I just published. It will also be included in a future release of pylance.
